internal/go_repository.bzl

```diff
@@ -212,7 +221,7 @@ def _go_repository_impl(ctx):
             existing_build_file = name
             break
```

Comment:
The loop that these are in (which I can't comment on entirely due to GitHub's UI) includes some use of `ctx.path`, so could potentially result in a restart.  Moving `build_file_paths = [ctx.path(name) for name in ctx.attr.build_file_name.split(",")]` to before the fetch step might help a lot.